main.go

```diff
@@ -88,108 +96,175 @@ func (c *DownloadPlugin) Run(cliConnection plugin.CliConnection, args []string)
+		// stop consoleWriter
+		quit := make(chan int)
 
-	// disable consoleWriter if verbose
-	if flagVals.Verbose_flag == false {
-		go consoleWriter(quit)
-	}
+		// disable consoleWriter if verbose
+		if flagVals.Verbose_flag == false {
+			go consoleWriter(quit)
+		}
 
-	// Start the download
-	wg.Add(1)
-	dloader.Download(files, dirs, startingPathServer, rootWorkingDirectoryLocal, filterList)
+		if flagVals.File_flag {
+			// create directory for single file
+			err := os.MkdirAll(strings.TrimSuffix(v.RootWorkingDirectoryLocal, filepath.Base(v.RootWorkingDirectoryLocal)), 0755)
+			check(err, "Error D1: failed to create directory.")
+
+			// start download of single file
+			wg.Add(1)
+			dloader.DownloadFile(v.StartingPathServer, v.RootWorkingDirectoryLocal, &wg)
+		} else {
+			// parse the input directory
+			files, dirs := parser.ExecParseDir(v.StartingPathServer)
+
+			// start download of directory
+			wg.Add(1)
+			dloader.Download(files, dirs, v.StartingPathServer, v.RootWorkingDirectoryLocal, filterList)
+		}
 
-	// Wait for download goRoutines
-	wg.Wait()
+		// wait for download goRoutines
+		wg.Wait()
 
-	// stop console writer
-	if flagVals.Verbose_flag == false {
-		quit <- 0
+		// stop console writer
+		if flagVals.Verbose_flag == false {
+			quit <- 0
+		}
 	}
 
+	// return completion status to user
 	getFailedDownloads()
 	PrintCompletionInfo(start, onWindows)
-
 }
 
 /*
-*	-----------------------------------------------------------------------------------------------
-* 	------------------------------------- Helper Functions ----------------------------------------
-* 	-----------------------------------------------------------------------------------------------
+*	---------------------------------------------------------------------------------------
+* 	--------------------------------- Helper Functions ------------------------------------
+* 	---------------------------------------------------------------------------------------
  */
 
+/*
+*	This function returns a list of files that failed to download.
+ */
 func getFailedDownloads() {
 	failedDownloads = append(parser.GetFailedDownloads(), dloader.GetFailedDownloads()...)
 }
 
-func GetDirectoryContext(workingDir string, copyOfArgs []string) (string, string) {
-	rootWorkingDirectory := workingDir + "/" + appName + "-download/"
+/*
+*	This function returns a list of pathVal structs that contain the locations
+*	to download each input path to and from.
+ */
+func GetDirectoryContext(workingDir string, paths []string, isFile bool) []pathVal {
+	var pathVals []pathVal
 
-	// append path if provided as arguement
+	rootWorkingDir := workingDir + "/"
+	localPath := rootWorkingDir
 	startingPath := "/"
-	if len(copyOfArgs) > 2 && !strings.HasPrefix(copyOfArgs[2], "-") {
-		startingPath = copyOfArgs[2]
-		if !strings.HasSuffix(startingPath, "/") {
-			startingPath += "/"
+
+	if len(paths) == 0 {
+		// create appName directory if downloading whole app
+		addPathVals := pathVal{
```

Comment:
Maybe don't even worry about `os.PathSeparator`, but call `filepath.FromSlash` on the `RootWorkingDirectoryLocal` field after you concatenate those strings?